src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -1,256 +1,119 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#nullable enable
+
 using System;
-using System.Collections.Generic;
-using System.Linq;
+using System.Collections.Immutable;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.CodeRefactorings;
-using Microsoft.CodeAnalysis.Editing;
-using Microsoft.CodeAnalysis.Formatting;
 using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Operations;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.ConvertIfToSwitch
 {
-    internal abstract class AbstractConvertIfToSwitchCodeRefactoringProvider : CodeRefactoringProvider
+    internal abstract partial class AbstractConvertIfToSwitchCodeRefactoringProvider : CodeRefactoringProvider
     {
-        public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        public override Task ComputeRefactoringsAsync(CodeRefactoringContext context)
         {
-            var document = context.Document;
-            if (document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles)
-            {
-                return;
-            }
-
-            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-            var semanticModel = await document.GetSemanticModelAsync().ConfigureAwait(false);
-
-            await CreateAnalyzer(syntaxFacts, semanticModel)
-                .ComputeRefactoringsAsync(context).ConfigureAwait(false);
+            var analyzer = CreateAnalyzer(context.Document.GetLanguageService<ISyntaxFactsService>());
+            return analyzer.ComputeRefactoringsAsync(context);
         }
 
-        protected interface IPattern<TSwitchLabelSyntax>
-            where TSwitchLabelSyntax : SyntaxNode
-        {
-            TSwitchLabelSyntax CreateSwitchLabel();
-        }
+        public abstract IAnalyzer CreateAnalyzer(ISyntaxFactsService syntaxFacts);
 
-        protected interface IAnalyzer
+        public interface IAnalyzer
         {
             Task ComputeRefactoringsAsync(CodeRefactoringContext context);
         }
 
-        protected abstract IAnalyzer CreateAnalyzer(ISyntaxFactsService syntaxFacts, SemanticModel semanticModel);
+        public abstract class Pattern
+        {
+        }
 
-        protected abstract class Analyzer<TStatementSyntax, TIfStatementSyntax, TExpressionSyntax, TSwitchLabelSyntax> : IAnalyzer
-            where TExpressionSyntax : SyntaxNode
-            where TIfStatementSyntax : SyntaxNode
-            where TSwitchLabelSyntax : SyntaxNode
+        public sealed class SwitchLabel
```

Comment:
consider comments.  i can't tell if this is supposed to represent the code that was anlayzed, the code that is being generated, (both), etc.